### PR TITLE
CIAC-8513 same reviewer for same contributor

### PR DESCRIPTION
**Pull request by edik24**
Change  flow of handle-external-pr

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-8513)

## Description
When a PR is opened check if the author already has an open PR, and notify the contribution TL.

This message will help us assign the same reviewer to different PRs from the same contributor, which can save work time (one demo for both PRs, for example), and provide a better experience for the contributor (only one reviewer).

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 1]
```diff
@@ -278,6 +277,97 @@ def is_tim_reviewer_needed(pr_files: list[str], support_label: str) -> bool:
...
+    Get all PR's that are currently open from the same author, filter the list and return reviewer if reviewer is part
+    of the current contribution round
+    Arguments:
+    - other_prs_by_same_user - list of opened PR's
+
+    Returns:
+    - Reviewer of the found pr's
+    """
+    content_roles = load_json(CONTENT_ROLES_PATH)
+    content_reviewers, __, _ = get_content_reviewers(content_roles)
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:336]: you don't need to call this function, you already have content_reviewers pulled in the main

**Review by ShahafBenYakir — COMMENTED** [review]
Good job! 
Sent you some more feedback on priavte.

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 2]
```diff
@@ -278,6 +277,97 @@ def is_tim_reviewer_needed(pr_files: list[str], support_label: str) -> bool:
...
+    Arguments:
+    - pr - the opened PR
+
+    Returns:
+    - Found User
+    """
+    body = pr.body
+    index_of_user = body.find("Contributor\n@")
+    substring_user = body[index_of_user + len("Contributor\n@"):]
+    user_in_list = substring_user.split("\n")[0]
```
[the remark this answers is not in the document]
Comment (edik24) [reply Utils/github_workflow_scripts/handle_external_pr.py:292]: I've tested what you suggested, it worked for me when I changed the 2nd line:
PR_AUTHOR_PATTERN = '## Contributor\n@(.*)'
matcher = re.search(PR_AUTHOR_PATTERN, body)
matcher.groups()[0]

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 3]
```diff
@@ -278,6 +277,97 @@ def is_tim_reviewer_needed(pr_files: list[str], support_label: str) -> bool:
...
+    content_reviewers, __, _ = get_content_reviewers(content_roles)
+    reviewers = []
+    relevant_reviewer = ''
+    for pr in other_prs_by_same_user:
+        for reviewer in pr.requested_reviewers:
+            print(f'reviewer of the pr is: {reviewer.login}')
+            reviewers.append(reviewer.login)
+        for reviewer in reviewers:
+            if reviewer in content_reviewers:
+                relevant_reviewer = reviewer
```
[the remark this answers is not in the document]
Comment (edik24) [reply Utils/github_workflow_scripts/handle_external_pr.py]: I changed it the way you proposed at the beginning, but the pr.requested_reviewers is not a list of users.
It's a list of specific NameUser type from Github:
[NamedUser(login="anas-yousef")]

So anyway I need one for loop to create the reviewers logins list.
After that I used the offer you suggested

[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply Utils/github_workflow_scripts/handle_external_pr.py]: you can do `set(list(pr.requested_reviewers))`

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 4]
```diff
@@ -359,13 +353,21 @@ def find_reviewer_to_assign(content_repo, pr, pr_number):
...
-        pr_creator = get_user_from_ui_pr(pr)
+        pr_creator = get_user_from_pr_body(pr)
     else:
         pr_creator = pr.user.login
 
     other_prs_by_same_user = find_all_open_prs_by_user(content_repo, pr_creator, pr_number)
-    reviewers_to_assign = reviewer_of_prs_from_current_round(other_prs_by_same_user)
-    return reviewers_to_assign
+
+    reviewers_to_assign = reviewer_of_prs_from_current_round(other_prs_by_same_user, content_reviewers)
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:362]:
```suggestion
    reviewer_to_assign = reviewer_of_prs_from_current_round(other_prs_by_same_user, content_reviewers)
```

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 5]
```diff
@@ -332,22 +330,18 @@ def reviewer_of_prs_from_current_round(other_prs_by_same_user) -> str:
...
+        existing_reviewer = content_reviewers_set.intersection(reviewer_names)
+        if existing_reviewer:
+            return existing_reviewer.pop()
+        else:
+            print("There are other PR's by same author, but their reviewer is not in the current contribution round")
+    return ''
 
 
-def find_reviewer_to_assign(content_repo, pr, pr_number):
+def find_reviewer_to_assign(content_repo: Repository, pr: PullRequest, pr_number: str, content_reviewers: list[str]):
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:344]:
```suggestion
def find_reviewer_to_assign(content_repo: Repository, pr: PullRequest, pr_number: str, content_reviewers: list[str]) -> str:
```

File: Utils/github_workflow_scripts/handle_external_pr.py [hunk 6]
```diff
@@ -287,13 +289,14 @@ def get_user_from_ui_pr(pr) -> str:
     - Found User
     """
     body = pr.body
-    index_of_user = body.find("Contributor\n@")
-    substring_user = body[index_of_user + len("Contributor\n@"):]
-    user_in_list = substring_user.split("\n")[0]
-    return user_in_list
+    PR_AUTHOR_PATTERN = '## Contributor\n@(.*)'
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/handle_external_pr.py:292]: PR_AUTHOR_PATTERN should be a constant variable in the file top level.

**Review by ShahafBenYakir — APPROVED** [review]
Great, small leftovers

**Pull request merged**